File: C++/M01_ProgrammingBasics/L06_NestedLoops/Exercises/Solutions/P03_SumPrimeNonPrime.cpp
#include <string>
#include <iostream>
using namespace std;

int main() {
	string input;
	cin >> input;

	int sumPrime = 0; 
	int sumNonPrime = 0;

	while (input != "stop") {
		int number = stoi(input);

		if (number < 0) {
			cout << "Number is negative." << endl;
		} else {
			int count = 0;
			for (int diff = 1; diff <= number; diff++) {
				if (number % diff == 0) {
					count++;
				}
			}
			if (count == 2) {
				sumPrime += number;
			} else {
				sumNonPrime += number;
			}
		}
		
		cin >> input;
	}

	cout << "Sum of all prime numbers is: " << sumPrime << endl;
	cout << "Sum of all non prime numbers is: " << sumNonPrime << endl;

	return 0;
}

File: Python/M01_ProgrammingBasics/L06_NestedLoops/Exercises/Solutions/P03_SumPrimeNonPrime.py
command = input()

prime_number = 0
non_prime_number = 0

while command != "stop":
    number = int(command)
    delitel = 0
    if number < 0:
        print("Number is negative.")
    else:
        for x in range(2, number + 1):
            if number % x == 0:
                delitel += 1
    if delitel == 1:
        prime_number += number
    elif delitel > 1:
        non_prime_number += number

    command = input()

print(f"Sum of all prime numbers is: {prime_number}")
print(f"Sum of all non prime numbers is: {non_prime_number}")